### PR TITLE
feat: [CDS-83171]: webhooks product / customer feedback

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

-  Make folder paths optional and added the helper text below.
-  Increased the width of the payload drawer.
-  Highlight on the `webhook` name in create / edit and delete flow.
- Added concerned tooltips in the `new webhook modal`
- Made the payloads drawer light theme and changed the background to white
- Added search on eventId for webhook events page
- Added search on webhook sync drawer
- Added background color on the selected row in webhook sync drawer

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/c0ac8809-036c-4a74-94a8-610626b1e1c2

https://github.com/harness/harness-core-ui/assets/106532291/4088b38c-d671-44ac-807d-38d185ee2803

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by vikrant-harness** [comment]
Automation Ticket :- https://harness.atlassian.net/browse/CDS-78807

File: src/modules/70-pipeline/pages/webhooks/WebhookEvents/WebhookEvents.tsx [hunk 1]
```diff
@@ -107,6 +109,12 @@ export default function WebhookEvents(): JSX.Element {
       }
     })
   }
+  const onSearchChange = (eventIdentifier: string): void => {
+    updateQueryParams({
+      page: 0,
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/pages/webhooks/WebhookEvents/WebhookEvents.tsx:114]:
```suggestion
      page: DEFAULT_PAGE_INDEX,
```

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/70-pipeline/pages/webhooks/NewWebhookModal.tsx [hunk 2]
```diff
@@ -174,6 +163,7 @@ export default function NewWebhookModal(props: NewWebhookModalProps): JSX.Elemen
                   <ConnectorReferenceField
                     name="connectorRef"
                     width={353}
+                    tooltipProps={{ dataTooltipId: 'webhook-connectorRef' }}
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/webhooks/NewWebhookModal.tsx:166]: Just wondering, can this be common id that we use at other places? Connector has same meaning everywhere. So, are we going to add some custom text for this one?

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by PratyushGar — DISMISSED** [review]

File: src/modules/70-pipeline/pages/webhooks/WebhookEvents/WebhookEvents.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/pages/webhooks/WebhookEvents/WebhookEvents.tsx:114]: will update in next PR

File: src/modules/70-pipeline/pages/webhooks/NewWebhookModal.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/pages/webhooks/NewWebhookModal.tsx:166]: yes UX team wanted specific tooltips for this modal

**Review by srdjan-harness — APPROVED** [review]

File: src/modules/70-pipeline/pages/webhooks/WebhookEvents/WebhookEvents.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/pages/webhooks/WebhookEvents/WebhookEvents.tsx]: Done

**Review by PratyushGar — DISMISSED** [review]

**Review by srdjan-harness — DISMISSED** [review]

**Comment by vikrant-harness** [comment]
Automation ticket :- https://harness.atlassian.net/browse/CDS-78807 :)

**Comment by kalyansai-harness** [comment]
@vikrant-harness why on search by event id is hiding all the content? Only the results has to get updated rite? why are we removing the search bar also?

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Review by srdjan-harness — APPROVED** [review]

**Pull request merged**